apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -1199,6 +1202,30 @@ type FIPSConfig struct {
 	CustomFIPSConfig *CustomConfig `json:"customFIPSConfig,omitempty"`
 }
 
+type AgentSidecarInjectionConfig struct {
+	// Enabled enables Sidecar injections.
+	// Default: true
+	// +optional
+	Enabled *bool `json:"enabled"`
+	// ClusterAgentCommunicationEnabled enables communication between Agent sidecars and the Cluster Agent.
+	// Default : true
```

Comment:
I think such defaults (and empty structs) should be initialized somewhere here
https://github.com/DataDog/datadog-operator/blob/f0e5f6be4421603604481d94e9fd3c30b37723c2/apis/datadoghq/v2alpha1/datadogagent_default.go#L72-L74

@celenechang could you please confirm?